azdev/operations/tests/pytest_runner.py

```diff
@@ -4,16 +4,24 @@
 # license information.
 # -----------------------------------------------------------------------------
 
+import os
+
+from knack.log import get_logger
+
+from azdev.utilities import call
+
 
 def get_test_runner(parallel, log_path, last_failed):
     """Create a pytest execution method"""
     def _run(test_paths, pytest_args):
-        from azdev.utilities import call
-        from knack.log import get_logger
 
         logger = get_logger(__name__)
 
-        arguments = ['-p', 'no:warnings', '--no-print-logs', '--junit-xml', log_path]
+        if os.name == 'posix':
+            arguments = ['-x', '-v', '--boxed', '-p no:warnings', '--log-level=WARN', '--junit-xml', log_path]
```

Comment:
The added `-x` argument means:

>   -x, --exitfirst       Exit instantly on first error or failed test
> https://docs.pytest.org/en/7.3.x/reference/reference.html#command-line-flags

This is a breaking change to the original behavior.